sched/irq/irq_csection.c

```diff
@@ -290,6 +290,9 @@ irqstate_t enter_critical_section(void)
                   DEBUGVERIFY(up_cpu_paused(cpu));
                   paused = true;
 
+                  /* After resume current_task may change */
+
+                  up_update_task(current_task(cpu));
```

Comment:
After resume, do we already know which task is going to be scheduled ?

Reply:
current_task(this_cpu()) is the running task going to be scheduled